app/pages/reset-password.spec.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import assert from 'assert';
+import { shallow } from 'enzyme';
+import ResetPasswordPage from './reset-password';
+
+const user = {
+  display_name: 'Test User'
+};
+
+const location = {
+  query: {
+    reset_password_token: 'test'
+  }
+};
+
+describe('ResetPasswordPage', () => {
+  let wrapper;
+
+  before(() => {
+    wrapper = shallow(<ResetPasswordPage />, { context: { router: { push: () => {} }}});
+  });
+
+  it('should render without crashing', () => {
+    const ResetPasswordPageContainer = wrapper.find('div.centered-grid');
+    assert.equal(ResetPasswordPageContainer.length, 1);
+  });
+
+  describe('#renderSubmitEmailForm', () => {
+    it('does not allow a signed in user to submit their email', () => {
+      wrapper.setProps({ user });
+      assert.equal(wrapper.find('Translate').prop('content'), 'resetPassword.loggedInDialog');
+    });
+
+    it('renders an email input form for a not logged in user', () => {
```

Comment:
When you extract out the forms into their own functional components you can rework these tests to check for the presence of the correct form.